### ISSUE TITLE
Overfitting on subset of data with 2D bounding boxes

### ISSUE THREAD

**Issue by mathild7** [waymo-research/waymo-open-dataset#76]
Hi,

Has anyone played with the subset of data containing natively annotated images? I'm currently experimenting with it and am getting overfitting after 3-4 epoch. I am using a Faster-RCNN base architecture and have experimented with dropout (made the results worse).

The best accuracy I have gotten thus far on the validation set is ~65%. (Daytime only)

I should not need to un-distort the image as the bounding boxes are natively in that frame of reference, correct?

Thanks!

**Comment by peisun1115** [comment]
We have released a paper which has 2D baseline.

https://arxiv.org/pdf/1912.04838.pdf

**Comment by mathild7** [comment]
> We have released a paper which has 2D baseline.
> 
> https://arxiv.org/pdf/1912.04838.pdf

Hi,

Thanks for the reply. A few follow up questions
1) Does the baseline include some forms of data augmentation?
2) Do the metric results include night time and dusk/dawn?

Thanks
-Mat

**Comment by henrikkretzschmar** [comment]
The camera object detection and tracking baselines that we present in the paper are trained on native camera labels for all 1k run segments without any data augmentation. FYI, we are planning on publicly releasing those camera labels for all 1k run segments within the next few weeks.

Training on the larger dataset should reduce any overfitting issues that you may be observing in your experiments. However, we have successfully trained object detectors on the subset of data that is already public without any data augmentation. Do you shuffle the data samples for training?

Thanks!

**Comment by mathild7** [comment]
> The camera object detection and tracking baselines that we present in the paper are trained on native camera labels for all 1k run segments without any data augmentation. FYI, we are planning on publicly releasing those camera labels for all 1k run segments within the next few weeks.
> 
> Training on the larger dataset should reduce any overfitting issues that you may be observing in your experiments. However, we have successfully trained object detectors on the subset of data that is already public without any data augmentation. Do you shuffle the data samples for training?
> 
> Thanks!

Hi,

What do you mean by the 1k run segments? Currently I am using the subset of data available from your website. Do you mean to say all images currently available will soon have native 2D bounding boxes?

I do shuffle my images in my training set every epoch.
What do you mean by successfully trained? I am curious as to how many epoch it took for your implementation to 'converge'. This was with a Faster RCNN implementation as well?

Thanks again,
-Mat

**Comment by mathild7** [comment]
> The camera object detection and tracking baselines that we present in the paper are trained on native camera labels for all 1k run segments without any data augmentation. FYI, we are planning on publicly releasing those camera labels for all 1k run segments within the next few weeks.
> 
> Training on the larger dataset should reduce any overfitting issues that you may be observing in your experiments. However, we have successfully trained object detectors on the subset of data that is already public without any data augmentation. Do you shuffle the data samples for training?
> 
> Thanks!

Hi there
Following up. Still seeing similar problems. When do you plan on releasing all 1k runs worth of 2d BBox labels? Do you have more information on the training you did with the subset of labels?

Thanks!

**Comment by peisun1115** [comment]
We are planning to release the remaining 2D labels in our next patch release. It will be soon. I cannot promise the exactly time.

**Comment by mathild7** [comment]
> We are planning to release the remaining 2D labels in our next patch release. It will be soon. I cannot promise the exact time.

Perfect thanks I'll keep my eyes peeled. Do you have more info on what Henrik was saying with the current subset?

**Comment by peisun1115** [comment]
1. The model we used is FasterRCNN
2. What Henrik meant was that even with only 100 scenes (released), he was able to achieve reasonable AP while testing on a hold-out set. I do not know whether there was overfitting issue as you have seen. And I do not know what was the AP on the hold-out set. Henrik can comment more.

**Comment by carranza96** [comment]
Could Henrik provide more details on how the faster-rcnn model was trained for camera object detection and the results obtained with the subset of data? Thanks

**Comment by henrikkretzschmar** [comment]
For context: while our initial Waymo Open Dataset release contained 1k segments, we only provided camera labels for 100 of these segments.

We're currently targeting to release the camera labels for all 1k segments by next week -- I can't promise it, but we're trying to make it happen. The larger dataset will hopefully mitigate any issues that folks may have been experiencing in regards to overfitting.

One of the models that we trained a while ago on the 100 segments for which we have already released the camera labels was a Faster R-CNN model with ResNet-101. It looks like we got ~0.4 AP on the training set and ~0.3 AP on the validation set in one particular experiment. I don't think that we did a lot of tuning for this experiment, though. Note that this is not the same model that we used for the baselines in the paper.

**Comment by mathild7** [comment]
> For context: while our initial Waymo Open Dataset release contained 1k segments, we only provided camera labels for 100 of these segments.
> 
> We're currently targeting to release the camera labels for all 1k segments by next week -- I can't promise it, but we're trying to make it happen. The larger dataset will hopefully mitigate any issues that folks may have been experiencing in regards to overfitting.
> 
> One of the models that we trained a while ago on the 100 segments for which we have already released the camera labels was a Faster R-CNN model with ResNet-101. It looks like we got ~0.4 AP on the training set and ~0.3 AP on the validation set in one particular experiment. I don't think that we did a lot of tuning for this experiment, though. Note that this is not the same model that we used for the baselines in the paper.

Thanks a lot for your reply, that makes me feel a lot better about the results I was seeing. Can't wait to see all segments, I really believe this will reduce the overfitting I was seeing.

**Comment by carranza96** [comment]
@henrikkretzschmar thanks for the info.
I am now training with the complete 2D images dataset. Although mAP has improved, I am still unable to get the same results as the baseline presented in the paper. I am finetuning the FasterRCNN model pretrained with the COCO dataset, which is what you mention in the paper.

Currently, I am obtaining:

OBJECT_TYPE_TYPE_VEHICLE_LEVEL_1: [mAP 0.603818] [mAPH 0.603818]
OBJECT_TYPE_TYPE_VEHICLE_LEVEL_2: [mAP 0.497749] [mAPH 0.497749]
OBJECT_TYPE_TYPE_PEDESTRIAN_LEVEL_1: [mAP 0.580123] [mAPH 0.580123]
OBJECT_TYPE_TYPE_PEDESTRIAN_LEVEL_2: [mAP 0.53047] [mAPH 0.53047]

Still far from the 68.4 mAP of vehicles, but better results on pedestrian detection.
I would be grateful if it is possible to know more details about the finetuning training. Image resolution, number of training steps, learning rate, etc.

Many thanks for your help.
